### ISSUE TITLE
`.path` on Uri is counterintuitive

### ISSUE THREAD

**Issue by kubukoz** [softwaremill/sttp-model#36]
Multiple calls to `.path` will overwrite each other, which is often unexpected, especially if you've been working with Tapir recently (which would append the input to the already defined ones):

```scala
baseUrl.path("v1", "api").path("demo") //results in just /demo
```

I would consider incrementally renaming `path` to `withPath` and adding a `addPath` to make it more obvious, as well as enable adding extra segments to an already existing path without going through `copy` manually.

**Comment by kubukoz** [comment]
As @adamw mentioned, this should also apply to query params and fragments, possibly headers as well?

**Comment by bwiercinski** [comment]
* presently **query parameters** can be appended :) `url.param("foo", bar).param("foo2", bar2)` works as expected
* for **fragment** this doesn't apply, because you cannot have multiple fragments in url
* **header** is a part of request not Uri :p (but Request Builder Api overrides existing header or appends as a new header)

**Comment by adamw** [comment]
Released in 1.2.0-RC5.

I've deprecated `.path` and `.params` and added `.[add|with][Path|Params]`
